### PR TITLE
new helm chart for the operator

### LINKED ISSUE #268: Add helm chart for operator
With migration from controller to operator, we removed the helm chart out of the operator. Add helm chart back again since it's required for ease of use.

### PR DESCRIPTION
**Pull request by magneticz**
As I said in my comment in the issue, I am not a helm expert, so it might not be the best chart you see, but I can update the chart with your comments, if there are any.

I just copied over old chart and made it work with new resource objects. Basically it's a derived from old helm chart and yaml files from /deploy folder in this repo. 

Closes #268

### CODE REVIEW

**Comment by stakater-user** [comment]
@magneticz Yikes! You better fix it before anyone else finds out! [Build 1](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-297/1/) has Failed!

**Comment by stakater-user** [comment]
@magneticz Yikes! You better fix it before anyone else finds out! [Build 2](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-297/2/) has Failed!

**Comment by magneticz** [comment]
Hi all, 
Any pointers on how to fix the failing build? I didn't touch any coding part, so not sure why it is failing at all. 
If there are any comments regarding the chart itself, would be happy to fix those as well.

**Comment by rasheedamir** [comment]
@josedev-union can you plz take a look?

**Comment by twardnw** [comment]
I'd also be willing to look at helping fix whatever is wrong, but without knowing what has failed, hard to do anything :( @josedev-union , what's the word?

**Comment by stakater-user** [comment]
@josedev-union Image is available for testing. `docker pull stakater/IngressMonitorController:SNAPSHOT-PR-297-32ed17aa`

**Comment by cuttingedge1109** [comment]
Sorry for late response.
let me check shortly

File: deploy/chart/ingressmonitorcontroller/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+name: imc-operator
+version: 1.0.50
+appVersion: 1.0.47
```
Comment (cuttingedge1109) [inline deploy/chart/ingressmonitorcontroller/Chart.yaml:4]: Set the appVersion as the latest release version `2.0.10`

File: deploy/chart/ingressmonitorcontroller/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+| rbac.create          | Option to create rbac resources                                                      | `true`                        | `true`
+| rbac.labels          | Labels for rbac                                                      | `{}`                        | `{}`
+| serviceAccount.create          | Option to create service account                                                      | `true`                        | `true`
+| serviceAccount.labels          | Labels for serviceAccount                                                      | `{}`                        | `{}`
+| serviceAccount.name          | Name of serviceAccount                                                      | `ingressmonitorcontroller`                        | `ingressmonitorcontroller`
+| image.name          | Image of ingressMonitorController                                                      | `stakater/ingressmonitorcontroller`                        | `stakater/ingressmonitorcontroller`
+| image.tag          | Version of ingressMonitorController Image                                                      | `1.0.47`                        | `1.0.47`
```
Comment (cuttingedge1109) [inline deploy/chart/ingressmonitorcontroller/README.md:37]: Please sync the default value with the values.yaml file.
`2.0.2` in values.yaml file now

File: deploy/chart/ingressmonitorcontroller/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+| image.pullPolicy          | Pull policy for image                                                      | `IfNotPresent`                        | `IfNotPresent`
+| providers.name          | Name of the provider                                                      | `UptimeRobot`                        | `UptimeRobot`                        |
+| providers.apiKey        | ApiKey of the provider                                                    | `u956-afus321g565fghr519`            | `your-api-key`                       |
```
Comment (cuttingedge1109) [inline deploy/chart/ingressmonitorcontroller/README.md:40]: Seems better to remove the default value for `providers.apiKey` in values.yaml file and set "" here.

File: deploy/chart/ingressmonitorcontroller/values.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    # The name of the ServiceAccount to use.
+    # If not set and create is true, a name is generated using the fullname template
+    name: imc-operator
+  image:
+    name: stakater/ingressmonitorcontroller
+    tag: 'v2.0.2'
+    pullPolicy: Always
+  providers:
+  - name: Uptime
+    apiKey: 0008465a8fdc6b5bb96ac30e7b6fdd278d20d346
```
Comment (cuttingedge1109) [inline deploy/chart/ingressmonitorcontroller/values.yaml:33]: I recommend to set empty string here

**Review by cuttingedge1109 — CHANGES_REQUESTED** [review]

**Comment by twardnw** [comment]
@magneticz , if you would like, I addressed the feedback in this commit : https://github.com/amazeeio/IngressMonitorController/commit/1cd85492798e003933f554d7242d1eabe69de662

**Comment by magneticz** [comment]
hey all, sorry for the delay, I was on vacation and was not working. 

@twardnw do you want to merge your changes in this PR?

**Comment by twardnw** [comment]
You should be able to add that repo as another remote and just merge the changes in

**Comment by Slutzky** [comment]
Hi, is this MR going to merged in near future ? thank you in advance

**Comment by magneticz** [comment]
hi all, sorry it took me so long to update the PR. I merged changes from @twardnw (thanks for addressing the comments).

**Comment by stakater-user** [comment]
@magneticz Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/IngressMonitorController/commit/793f50d4cf6c61c9a7c91e0e922041eb0a201305/checks) has Failed!

**Comment by Slutzky** [comment]
sorry to be annoying but is this MR will be closed ?

**Comment by rasheedamir** [comment]
@magneticz can you plz make the PR green? and then lets get it merged

**Comment by cwrau** [comment]
I'm also interested in this PR, can I help somehow?

**Comment by lgallard** [comment]
@magneticz @stakater-user any update with this PR implementation? I would be great having the chart back!

**Comment by ahmedwaleedmalik** [comment]
@kahootali is working on this in #315. It will released in a day or two. Apologies for the delay but I wasn't tracking this PR :)

**Comment by ahmedwaleedmalik** [comment]
`v2.0.15` is out now, helm chart is back. ✌️

**Comment by ahmedwaleedmalik** [comment]
Closing this PR now.

**Pull request closed without merging**